### PR TITLE
feat(codebuild): macOS codebuild support

### LINKED ISSUE #31170: codebuild: Add support for MacOS
### Describe the feature

CodeBuild recently added support for Mac builds. https://aws.amazon.com/about-aws/whats-new/2024/08/aws-codebuild-mac-builds/

The CDK can to be extended to support Mac CodeBuild fleets, as well as using those fleets in CodeBuild projects.

### Use Case

I have a couple of existing CodeBuild jobs created using CDK for building a project on AL2023 and Windows. I would like to extend this to Mac as well to ensure that builds work smoothly on as many platforms as possible.

### Proposed Solution

1) This [enum](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-codebuild/lib/environment-type.ts) needs to have an additional `MAC_ARM` value added to it.
2) A `MacBuildImage` class needs to be added to [this](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-codebuild/lib/project.ts) file.

Validation will also need to be added, since the MacOS support has some limitations. See the release article linked above for more information.

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.152.0

### Environment details (OS name and version, etc.)

Amazon Linux 2023

### PR DESCRIPTION
**Pull request by alanxluc**
### Issue # (if applicable)

Closes #31170.

### Reason for this change

CodeBuild recently added support for Mac builds. https://aws.amazon.com/about-aws/whats-new/2024/08/aws-codebuild-mac-builds/

The CDK can to be extended to support Mac CodeBuild fleets, as well as using those fleets in CodeBuild projects.

### Description of changes

- Added `MAC_ARM` value to EnvironmentType [enum](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-codebuild/lib/environment-type.ts)
- Added `MacBuildImage` to [project.ts](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-codebuild/lib/project.ts)

All design decisions were made by following the previous precedent set by `LinuxBuildImage` and `WindowsBuildImage`.

### Description of how you validated changes

I added unit tests and integration tests for `MacBuildImage`. I ensured that the CDK was able to build, and that the output CloudFormation stack had the correct resources. I also successfully built the Codebuild project that was deployed.

### Checklist
- [X] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by GavinZZ — APPROVED** [review]
Nice work! Excited to see this supported.

File: packages/aws-cdk-lib/aws-codebuild/lib/project.ts [hunk 1]
```diff
@@ -2240,6 +2241,112 @@ export class WindowsBuildImage implements IBuildImage {
...
+  public validate(buildEnvironment: BuildEnvironment): string[] {
+    const errors: string[] = [];
+
+    if (buildEnvironment.computeType && isLambdaComputeType(buildEnvironment.computeType)) {
+      errors.push('Mac images do not support Lambda compute types');
+    }
+
+    if (buildEnvironment.computeType != ComputeType.MEDIUM) {
+      errors.push('Mac images only support medium compute type');
+    }
```
Comment (moelasmar) [inline packages/aws-cdk-lib/aws-codebuild/lib/project.ts:2335]: @alanxluc .. I think this type of validation that there is a big probability to change, and I think it should not be part of CDK validations.

**Review by moelasmar — CHANGES_REQUESTED** [review]
I left one minor comment

File: packages/aws-cdk-lib/aws-codebuild/lib/project.ts (hunk 1 above)
Comment (alanxluc) [reply packages/aws-cdk-lib/aws-codebuild/lib/project.ts:2335]: Ack, will revise. Is the other validation good to keep?

Comment (moelasmar) [reply packages/aws-cdk-lib/aws-codebuild/lib/project.ts:2335]: yes, they look reasonable. Thanks @alanxluc for your contribution :)

**Review by moelasmar — APPROVED** [review]

**Pull request merged**